v0.14.6 release checklist

- [x] Shuffle any issues being put off to future milestones
- [x] Close all issues in current milestone (except this one)
- [x] Spring clean
    - [x] Re-fetch tooling
- [x] Configure and build
- [x] Pass all tests
    - [x] Remote CI
    - [x] Local
- [x] Cut release
- [x] Push release commit & tag to master
- Update website
    - [x] Copy and post manual, update 'latest' symlink and menu links
    - [x] Copy changelog and prefix with a summary as a blog post
    - [x] Tweak summary for gfm and edit into GitHub release notes
- Update downstream distro packages
	- [x] Arch Linux official: [community/sile](https://archlinux.org/packages/community/x86_64/sile/)
	- [x] Arch Linux AUR: [AUR/sile-luajit](https://aur.archlinux.org/packages/sile-luajit) <!-- – [pull request]() -->
    - [ ] Homebrew: [Formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/sile.rb) – [pull request](https://github.com/Homebrew/homebrew-core/pull/118254)
    - [ ] NixOS: [nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/typesetting/sile/default.nix) – [pull request](https://github.com/NixOS/nixpkgs/pull/206284) – [tracker](https://nixpk.gs/pr-tracker.html?pr=206284)
        - [ ] Drop devel flag from any website examples that were using it
    - [x] Ubuntu: [ppa](https://launchpad.net/~sile-typesetter/+archive/ubuntu/sile)
    - [x] Docker Hub: [tags](https://hub.docker.com/repository/docker/siletypesetter/sile/tags)
    - [x] GHCR: [versions](https://github.com/orgs/sile-typesetter/packages/container/sile/versions)
    - [x] openSUSE: pinging @Pi-Cla re [specfile](https://build.opensuse.org/package/show/openSUSE%3AFactory/sile)
    - [ ] Fedora: pinging @hegjon re [copr](https://copr.fedorainfracloud.org/coprs/jonny/SILE/)
    - [ ] NetBSD: pinging @jsonn re [pkgsrc](https://pkgsrc.se/print/sile)
    - [ ] Void Linux: [pull request](https://github.com/void-linux/void-packages/pull/18306)
    - [ ] OpenBSD: [mailing list thread](https://marc.info/?t=157907840200001)
- Bump downstream projects
    - [ ] [FontProof](https://github.com/sile-typesetter/fontproof) (Docker image base plus CI matrix)
    - [ ] [CaSILE](https://github.com/sile-typesetter/casile) (check, consider patch release)
- [ ] Eat cake

[![Packaging status](https://repology.org/badge/vertical-allrepos/sile.svg)](https://repology.org/project/sile/versions)

@hegjon It looks like the COPR build got hung up finding Genitum Plus:

```
checking whether font family Gentium Plus is available... configure: error: in `/builddir/build/BUILD/sile-0.14.6':
configure: error: font family Gentium Plus not found
```

I bumped into this packaging for Homebrew too. Unfortunately right now we have an all or nothing `--dilable-dependency-checks` which will work around this but at the cost of not doing lots of other useful checks. I'll look into something else in the future, but for this release if the font is not available at build time (and their is a plan to have it available at run time) probably the best thing to do is just patch out the `AX_FONT` line in `configure.ac`. For future releases maybe we can improve on that to not lump strongly suggested dependencies in with hard dependencies.

--- a/.github/workflows/nix.yml
+++ b/.github/workflows/nix.yml
@@ -19,7 +19,15 @@ jobs:
         with:
           extra_nix_config: |
             access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+      # Upstream package sometimes has flags set that disable flake checking
+      - name: Setup test env
+        run: |
+          echo 'NIXPKGS_ALLOW_BROKEN=1' >> $GITHUB_ENV
+      - name: Check flake
+        run: nix flake check --impure
       - name: Build flake
         run: nix build
-      - name: Test run flake
+      - name: Test run in shell
         run: nix shell -c sile --version
+      - name: Test run as flake
+        run: nix run . -- --version

--- a/Makefile-fonts
+++ b/Makefile-fonts
@@ -1,3 +1,5 @@
+if FONT_DOWNLOAD_TOOLS
+
 .fonts: fonttooling
 	[ -h .fonts ] || mkdir -p $@
 
@@ -6,8 +8,8 @@
 
 .PHONY: fonttooling
 fonttooling:
-	$(if $(BSDTAR),,$(error Please set BSDTAR with path or ./configure --with-manual or --enable-developer))
-	$(if $(CURL),,$(error Please set CURL with path or ./configure --with-manual or --enable-developer))
+	$(if $(BSDTAR),,$(error Please set BSDTAR with path or `./configure --enable-developer`))
+	$(if $(CURL),,$(error Please set CURL with path or `./configure --enable-developer`))
 
 # Don't let make delete things we had to download
 .PRECIOUS: .fonts/% .sources/%
@@ -136,3 +138,5 @@ notobase = $(shell echo $(notdir $1) | sed -e 's/-.*//')
 .fonts/Symbola%: .sources/Symbola.zip | .fonts
 	$(BSDTAR) -x -f $< -C $(dir $@) $(notdir $@) && \
 	touch $@
+
+endif

--- a/build-aux/ax_font.m4
+++ b/build-aux/ax_font.m4
@@ -8,10 +8,14 @@ AC_DEFUN([AX_FONT], [
     fi
     pushdef([FONT],$1)
     AC_MSG_CHECKING(whether font family FONT is available)
-    AS_IF([$FCMATCH "FONT" family | $GREP -qx "FONT"],[
-        AC_MSG_RESULT(yes)
+    AS_IF([test "$FCMATCH" = "true"],[
+            AC_MSG_RESULT(skip)
     ],[
-        AC_MSG_FAILURE([font family FONT not found])
+        AS_IF([$FCMATCH "FONT" family | $GREP -qx "FONT"],[
+            AC_MSG_RESULT(yes)
+        ],[
+            AC_MSG_FAILURE([font family FONT not found])
+        ])
     ])
     popdef([FONT])
 ])dnl

--- a/configure.ac
+++ b/configure.ac
@@ -55,6 +55,9 @@ AM_COND_IF([SYSTEM_LIBTEXPDF],
                          [AC_MSG_FAILURE([--with-system-libtexpdf was given, but test for libtexpdf failed])]) ],
            [AC_CONFIG_SUBDIRS([libtexpdf])])
 
+AM_CONDITIONAL([FONT_DOWNLOAD_TOOLS], [test -z ${DEVELOPER_TRUE} || (test -z ${MANUAL_TRUE} && test -z ${IS_SDIST_FALSE})])
+AC_SUBST([FONT_DOWNLOAD_TOOLS])
+
 AM_COND_IF([DEPENDENCY_CHECKS], [
 
     AX_FONT(Gentium Plus)
@@ -159,8 +162,7 @@ AM_COND_IF([DEPENDENCY_CHECKS], [
 
     # Required for downloading fonts for the manual and for tests
     # Since the source tarball includes a prebuilt manual we only need this for Git source builds
-    AM_CONDITIONAL([FONT_DOWNLOAD_TOOLS], [test -z ${DEVELOPER_TRUE} || (test -z ${MANUAL_TRUE} && test -z ${IS_SDIST_FALSE})])
-    AS_IF([FONT_DOWNLOAD_TOOLS], [
+    AM_COND_IF([FONT_DOWNLOAD_TOOLS], [
         AX_PROGVAR([curl])
         AX_PROGVAR([bsdtar])
     ])

--- a/configure.ac
+++ b/configure.ac
@@ -158,14 +158,14 @@ AM_COND_IF([DEPENDENCY_CHECKS], [
     ])
 
     # Required for downloading fonts for the manual and for tests
-    AM_COND_IF([MANUAL], [
+    # Since the source tarball includes a prebuilt manual we only need this for Git source builds
+    AM_CONDITIONAL([FONT_DOWNLOAD_TOOLS], [test -z ${DEVELOPER_TRUE} || (test -z ${MANUAL_TRUE} && test -z ${IS_SDIST_FALSE})])
+    AS_IF([FONT_DOWNLOAD_TOOLS], [
         AX_PROGVAR([curl])
         AX_PROGVAR([bsdtar])
     ])
 
     AM_COND_IF([DEVELOPER], [
-        AX_PROGVAR([curl])
-        AX_PROGVAR([bsdtar])
         AX_PROGVAR([busted])
         AX_PROGVAR([luacheck])
         AX_PROGVAR([luarocks])

--- a/flake.nix
+++ b/flake.nix
@@ -38,7 +38,7 @@
       version_rev = if (self ? rev) then (builtins.substring 0 7 self.rev) else "dirty";
       # Prepare a different luaEnv to be used in the overridden expression,
       # this is also the place to choose a different lua interpreter, such as
-      # lua5_3 or luajit
+      # lua5_4 or luajit
       luaEnv = pkgs.lua5_3.withPackages(ps: with ps; [
         cassowary
         cldr
@@ -117,11 +117,13 @@
           inherit (sile) checkInputs nativeBuildInputs;
           buildInputs = sile.buildInputs ++ [
             pkgs.libarchive
+            pkgs.perl
+          ];
+          luaEnv = luaEnv + pkgs.lua5_3.withPackages(ps: with ps; [
             pkgs.lua53Packages.busted
             pkgs.lua53Packages.luacheck
             pkgs.lua53Packages.luarocks
-            pkgs.perl
-          ];
+          ]);
         };
       };
       packages.sile = sile;

